### PR TITLE
[IMP] base/saas~18.1: remove res.partner.title model and field

### PR DESCRIPTION
**Pull request by malb-odoo**
The enterprise and community PR does a lot of changes on the partner form. One of the change was to remove the title field and so the model. This commit will adapt the tests accordingly

see: odoo/odoo#187357
see: odoo/enterprise#73879

task: 4306429

### CODE REVIEW

File: src/base/tests/test_util.py [hunk 1]
```diff
@@ -1268,31 +1268,31 @@ def test_ensure_xmlid_match_record(self):
 
     @unittest.skipUnless(util.version_gte("16.0"), "Only work on Odoo >= 16")
     def test_replace_in_all_jsonb_values(self):
-        test_partner_title = self.env["res.partner.title"].create(
+        test_partner_category = self.env["res.partner.category"].create(
```
Comment (aj-fuentes) [inline src/base/tests/test_util.py:1271]: The column must be jsonb, aka a translated field. I'd propose `res.country.group.name`

File: src/base/tests/test_util.py [hunk 2]
```diff
@@ -1060,8 +1060,8 @@ def test_rename_xmlid(self):
 
     def test_update_record_from_xml(self):
         # reset all fields on a <record>
-        xmlid = "base.res_partner_title_mister"
-        data_after = {"name": "Fortytwo", "shortcut": "42"}
+        xmlid = "base.res_partner_industry_A"
```
Comment (aj-fuentes) [inline src/base/tests/test_util.py:1063]: same here, `res.country.group`?

**Review by aj-fuentes — COMMENTED** [review]
I think `res.country.group` is a better replacement for `res.partner.title` from what I could check it has all the required columns and it's something we can inject and remove easily without conflicts with other data.

**Comment by aj-fuentes** [comment]
Add a reference to the upgrade PR.

File: src/base/tests/test_util.py [hunk 3]
```diff
@@ -1060,8 +1060,8 @@ def test_rename_xmlid(self):
...
+        data_after = {"name": "New Agriculture", "full_name": "New A - AGRICULTURE, FORESTRY AND FISHING"}
```
Comment (aj-fuentes) [inline src/base/tests/test_util.py:1064]: The idea here is to write something different from the standard.
```suggestion
        data_after = {"name": "42", "full_name": "Fortytwo"}
```

File: src/base/tests/test_util.py [hunk 4]
```diff
@@ -1158,14 +1158,14 @@ def test_update_record_translations_from_xml(self):
...
-        xmlid = "base.res_partner_title_mister"
+        xmlid = "base.res_partner_industry_A"
         util.update_record_from_xml(self.env.cr, xmlid, reset_translations=True)
 
         # change the translations to something arbitrary for all installed languages
         langs = self.env["res.lang"].get_installed()
         filter_lang = [code for code, _ in langs]
         self.assertIn(be_lang.code, filter_lang)
-        data_after = {"name": "Fortytwo", "shortcut": "42"}
+        data_after = {"name": "New Agriculture", "full_name": "New A - AGRICULTURE, FORESTRY AND FISHING"}
```
Comment (aj-fuentes) [inline src/base/tests/test_util.py:1168]: ditto

File: src/base/tests/test_util.py [hunk 5]
```diff
@@ -305,10 +305,10 @@ def test_any_operator(self):
         self.assertEqual(new_domain, expected)
 
         # test it also works recursively
-        domain = [("partner_id", "any", [("title", "not any", [("shortcut", "like", "S.A.")])])]
-        expected = [("partner_id", "any", [("title", "not any", [("abbr", "like", "S.A.")])])]
+        domain = [("partner_id", "any", [("bank_ids", "not any", [("shortcut", "like", "S.A.")])])]
+        expected = [("partner_id", "any", [("bank_ids", "not any", [("abbr", "like", "S.A.")])])]
```
Comment (aj-fuentes) [inline src/base/tests/test_util.py:309]: Use valid fields.
```suggestion
        domain = [("partner_id", "any", [("bank_ids", "not any", [("acc_number", "like", "S.A.")])])]
        expected = [("partner_id", "any", [("bank_ids", "not any", [("acc_nbr", "like", "S.A.")])])]
```

File: src/base/tests/test_util.py [hunk 6]
```diff
@@ -305,10 +305,10 @@ def test_any_operator(self):
...
 
-        new_domain = _adapt_one_domain(self.cr, "res.partner.title", "shortcut", "abbr", "res.company", domain)
+        new_domain = _adapt_one_domain(self.cr, "res.partner.bank", "shortcut", "abbr", "res.company", domain)
```
Comment (aj-fuentes) [inline src/base/tests/test_util.py:311]: valid field
```suggestion
        new_domain = _adapt_one_domain(self.cr, "res.partner.bank", "acc_number", "acc_nbr", "res.company", domain)
```

File: src/base/tests/test_util.py [hunk 7]
```diff
@@ -700,22 +700,22 @@ def test_explode_mult_filters(self):
     def test_explode_query_range(self):
         cr = self.env.cr
 
-        cr.execute("SELECT count(id) FROM res_partner_title")
+        cr.execute("SELECT count(id) FROM res_partner")
```
Comment (aj-fuentes) [inline src/base/tests/test_util.py:703]: Cannot use res_partner here. It fails in 12.3
```
ERROR: test_explode_query_range (odoo.addons.base.maintenance.migrations.base.tests.test_util.TestPG)
...
psycopg2.IntegrityError: update or delete on table "res_partner" violates foreign key constraint "res_partner_commercial_partner_id_fkey" on table "res_partner" 
```

Maybe keep using `res_partner_bank`?

Comment (malb-odoo) [reply src/base/tests/test_util.py]: Used res_partner_category to avoid creating a partner for the bank

**Review by aj-fuentes — APPROVED** [review]
@robodoo r+

**Comment by malb-odoo** [comment]
Had a conflict that broke some stuff, should be good now :grimacing:

**Comment by aj-fuentes** [comment]
Already approved ;)

**Pull request closed without merging**